### PR TITLE
Remove --color=true

### PR DESCRIPTION
**Pull request by paladox**
Reason since you can either use --color=false to disable colors or use colors which should be on by default without needing to use a config.

1.x release broke grunt see https://github.com/gruntjs/grunt-legacy-log/pull/16

So instead if it detect that --color=false or --no-color are not be using then it will switch colors on by default like it use too.

### CODE REVIEW

File: lib/system/supports-colors.js [hunk 1]
```diff
@@ -58,4 +57,4 @@ module.exports = (function () {
   }
 
   return false;
-})();
```
Comment (thiemowmde) [inline lib/system/supports-colors.js]: Please avoid unrelated changes like this.

Comment (paladox) [inline lib/system/supports-colors.js]: Hi sorry.

File: lib/system/supports-colors.js [hunk 2]
```diff
@@ -31,9 +31,8 @@ module.exports = (function () {
     return false;
   }
 
-  if (argv.indexOf('--color') !== -1 ||
-    argv.indexOf('--color=true') !== -1 ||
-    argv.indexOf('--color=always') !== -1) {
+  if (!argv.indexOf('--no-color') !== -1 ||
+    !argv.indexOf('--color=false') !== -1) {
```
Comment (thiemowmde) [inline lib/system/supports-colors.js]: This is wrong for multiple reasons.

1. The `||` is an or, which means this will always enable color when either the first _or_ the second argument is missing. This is _always_ the case, except when you specify _both_ arguments, which does not make much sense. I assume this was meant to be an `&&`.
2. With an `&&` this is the exact inverse of the `if` above. Which means all the code below this second `if` can never be reached.

Look at the code below. The issue you are experiencing may be hidden there.

I suggest to close this patch.

Comment (paladox) [inline lib/system/supports-colors.js]: Hi this was just a copy of the above code except from I added ! so if --no-color is used color is disabled if it is not used then color is enabled by default.

Comment (ricordisamoa) [inline lib/system/supports-colors.js]: `!argv.indexOf('--no-color')` is always a boolean, so it never equals `-1`

Comment (thiemowmde) [inline lib/system/supports-colors.js]: This is not how you invert an logical condition. I already described it in detail above.

Comment (paladox) [inline lib/system/supports-colors.js]: Ok thanks, your right I looked below and found this pull https://github.com/Marak/colors.js/pull/154 which fixed it.

**Comment by paladox** [comment]
In favour of https://github.com/Marak/colors.js/pull/154 which fixed it.
